assets/src/edit-story/app/media/useMediaReducer/reducer.js

```diff
@@ -92,6 +94,7 @@ function reducer(state, { type, payload }) {
       if (mediaType === state.mediaType) return state;
       return {
         ...INITIAL_STATE,
+        media: state.media.filter(({ local }) => local), // This filter allows remove temporary file returned on upload
```

Comment:
Why is this needed?

Reply:
When the user sets some Media Type Filter, it will call the Media API using [this declaration](https://github.com/google/web-stories-wp/blob/master/assets/src/edit-story/app/media/mediaProvider.js#L101-L103), which will always override the current media state (including local/temporary media), once it gets a response using [this method](https://github.com/google/web-stories-wp/blob/master/assets/src/edit-story/app/media/mediaProvider.js#L74-L80). 

Using this adjust, if I start to upload some media (consider a big file) and change the Media Type Filter, it reset media state to persist **only** local/temporary media, so when we have the response from the API here, [we just merge both data](https://github.com/google/web-stories-wp/blob/master/assets/src/edit-story/app/media/useMediaReducer/reducer.js#L51).

I can also add a `setMedia( { media: media.filter(({ local }) => local })` [here](https://github.com/google/web-stories-wp/blob/master/assets/src/edit-story/components/library/panes/media/mediaPane.js#L171-L173). WDYT? Have you some additional thoughts on it?